### PR TITLE
Gps service example

### PR DESCRIPTION
**Pull request by guilhermedemouraa**

### CODE REVIEW

File: py/examples/gps_client/__init__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Copyright (c) farm-ng, inc.
```
Comment (Hackerman342) [inline py/examples/gps_client/__init__.py:1]: rm this file

File: py/examples/gps_client/main.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                print(f"P DOP: {msg.p_dop}\n################################")
+
+        except AttributeError:
+            _outage_ctn[0] += 1
+            if _outage_ctn[0] > 0:
+                print(
+                    "Service is active but no messages are being published. "
+                    "Ensure your GPS antenna is unobstructed and "
+                    "try restarting the service."
+                )
```
Comment (Hackerman342) [inline py/examples/gps_client/main.py:70]: Needing this exception is a red flag for our service. It should still be sending the relposned proto, so the fields should still exist. Just they will be zero.

**Review by Hackerman342 — APPROVED** [review]

File: py/examples/gps_client/service_config.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
     "name": "gps",
     "port": 3001,
-    "host": "localhost",
+    "host": "lead-mango",
```
Comment (edgarriba) [inline py/examples/gps_client/service_config.json]: ?

File: py/examples/gps_client/main.py [hunk 4]
```diff
@@ -111,41 +32,49 @@ async def main(service_config_path: Path, msg_type: str) -> None:
...
-            print(f"Longitude: {gps_msg[3]}")
-            print(f"Altitude: {gps_msg[4]}")
-            print(f"Ground speed: {gps_msg[5]}")
-            print(f"Speed accuracy: {gps_msg[6]}")
-            print(f"Horizontal accuracy: {gps_msg[7]}")
-            print(f"Vertical accuracy: {gps_msg[8]}")
-            print(f"P dop: {gps_msg[9]}\n################################")
+    async for event, msg in EventClient(config).subscribe(config.subscriptions[0]):
+
+        try:
```
Comment (edgarriba) [inline py/examples/gps_client/main.py:37]: Why is this needed , are not the packages well formed?

Comment (guilhermedemouraa) [reply py/examples/gps_client/main.py:37]: Apparently, there are instances when car_soln = 0 that we receive packages without some of the variables, which causes the example to crash. We need to do some further investigation on the service. I will probably try to create a PR with steps to reproduce.

Comment (Hackerman342) [reply py/examples/gps_client/main.py:37]: @edgarriba We got to the bottom of it. The example was trying to parse the health / service metrics message as if it were a RelativePositionFrame proto because of the else
